api/v1alpha08/kogitoserverlessplatform_types.go

```diff
@@ -0,0 +1,157 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1alpha08
+
+import (
+	"github.com/kiegroup/container-builder/api"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ConfigurationSpec represents a generic configuration specification
+type ConfigurationSpec struct {
+	// Type represents the type of configuration, ie: property, configmap, secret, ...
+	Type string `json:"type"`
+	// Value the value to assign to the configuration (syntax may vary depending on the `Type`)
+	Value string `json:"value"`
+}
+
+const (
+	// KogitoServerlessPlatformKind is the Kind name of the KogitoServerlessPlatform CR
+	KogitoServerlessPlatformKind string = "KogitoServerlessPlatform"
+)
+
+// PlatformCluster is the kind of orchestration cluster the platform is installed into
+type PlatformCluster string
+
+const (
+	// PlatformClusterOpenShift is used when targeting an OpenShift cluster
+	PlatformClusterOpenShift PlatformCluster = "OpenShift"
+	// PlatformClusterKubernetes is used when targeting a Kubernetes cluster
+	PlatformClusterKubernetes PlatformCluster = "Kubernetes"
+)
+
+// PlatformBuildStrategy specifies how the Build should be executed.
+// It will trigger a Kaniko pod that will take care of producing the expected image containing the Quarkus/Kogito runtime.
+// +kubebuilder:validation:Enum=pod
+type PlatformBuildStrategy string
+
+const (
+	// BuildStrategyPod performs the build in a `Pod` (will schedule a new builder ephemeral `Pod` which will take care of the build action).
+	BuildStrategyPod PlatformBuildStrategy = "pod"
+)
+
+// PlatformBuildStrategies is a list of strategies allowed for the build
+var PlatformBuildStrategies = []PlatformBuildStrategy{
+	BuildStrategyPod,
+}
+
+// BuildPublishStrategies the list of all available publish strategies
+var BuildPublishStrategies = []api.PlatformBuildPublishStrategy{
+	api.PlatformBuildPublishStrategyKaniko,
+}
+
+// KogitoServerlessPlatformSpec defines the desired state of KogitoServerlessPlatform
+type KogitoServerlessPlatformSpec struct {
+	// Cluster what kind of cluster you're running (ie, plain Kubernetes or OpenShift)
+	Cluster PlatformCluster `json:"cluster,omitempty"`
+	// Build specify how to build the Workflow
+	Build api.BuildSpec `json:"build,omitempty"`
+	//BuildPlatform specify how it is the platform where we want to build the Workflow
+	BuildPlatform api.PlatformBuildSpec `json:"platform,omitempty"`
+	// Configuration list of configuration properties to be attached to all the Workflow built from this Platform
+	Configuration ConfigurationSpec `json:"configuration,omitempty"`
+}
+
+// PlatformPhase is the phase of an Platform
+type PlatformPhase string
+
+const (
+	// PlatformPhaseNone when the KogitoServerlessPlatform does not exist
+	PlatformPhaseNone PlatformPhase = ""
```

Comment:
I think we should use None here.

Reply:
Since the default value for a string is empty ("") in this way the None phase will be the default enum value.

If we put None here we have also to manage the empty or "" case that is the default one when a new platform is created.

Reply:
that's a good point, but "None" improves the readability.
it is not a mandatory change though.

Reply:
I think that in this way we will increase code readability at struct definition level but it will decrease when we use the PlatformPhase enum values because if we won't have an enum value mapping the empty string ("") at the first reconcile we have to compare it with the empty string instead of using one of the defined enum values.

If do you think that is it better we can change the name from None to Unknown (like here https://github.com/kubernetes/apimachinery/blob/57f2a0733447cfd41294477d833cce6580faaca3/pkg/apis/meta/v1/types.go#L734).